astropy/stats/funcs.py

```diff
@@ -790,3 +793,101 @@ def mad_std(data):
 
     # NOTE: 1. / scipy.stats.norm.ppf(0.75) = 1.482602218505602
     return median_absolute_deviation(data) * 1.482602218505602
+
+
+# TODO Note scipy dependency
+def sigma_to_logp(sigma, two_sided=True):
+    """Convert number-of-sigma to log-probability
+
+    Detection probabilities are often naturally expressed as a
+    false positive probability. On the other hand, astronomers
+    are used to thinking of small probabilities in terms of
+    how many standard deviations out on a normal distribution
+    they are. This function converts between these two
+    representations in a way that works even for probabilities
+    too small to represent in floating point (this happens at
+    about 40-sigma for double precision).
+
+    Parameters
+    ----------
+    sigma : array-like
+        The number of sigma
+    two_sided : boolean
+        If True (the default) use a two-tailed probability (probability
+        of the value being larger than sigma in absolute value); if
+        False use a one-sided probability
+
+    Returns
+    -------
+    logp : array-like
+        The (natural) logarithm of the corresponding probability
+
+    Examples
+    --------
+    >>> np.exp(sigma_to_logp(3))
+    0.0026997960632601926
+
+    >>> sigma_to_logp(50)/np.log(10)
+    -544.66530586633246
+    
+    """
+    import scipy.special
+    r = scipy.special.log_ndtr(-sigma)+(np.log(2) if two_sided else 0)
+    if two_sided:
+        if np.any(sigma<0):
+            raise ValueError("two-sided tail probability doesn't make sense for negative sigmas: %s" % sigma)
+    else:
+        if np.isscalar(sigma):
+            if sigma<0:
+                return np.log(scipy.stats.norm.sf(sigma))
+        else:
+            r[sigma<0] = np.log(scipy.stats.norm.sf(sigma[sigma<0]))
+    return r
+
+# TODO Note scipy dependency
+def logp_to_sigma(logp, two_sided=True):
+    """Convert log-probability to number-of-sigma
+
+    Detection probabilities are often naturally expressed as a
+    false positive probability. On the other hand, astronomers
+    are used to thinking of small probabilities in terms of
+    how many standard deviations out on a normal distribution
+    they are. This function converts between these two
+    representations in a way that works even for probabilities
+    too small to represent in floating point (this happens at
+    about 40-sigma for double precision). Both one- and two-tailed
+    versions are available.
+
+    Parameters
+    ----------
+    logp : float
```

Comment:
It would be very nice if this also worked for array-like inputs.
Is there a simple way to avoid having to write boilerplate code to loop over array entries?
Would [numpy.vectorize](http://docs.scipy.org/doc/numpy/reference/generated/numpy.vectorize.html) work well in this case?